SimplioOfficial/simplio-mobile
# Bug/24 refresh token is not correctly reused

# Overview

Refresh token is rotated on each its generation. Client did not paused http calls and refreshed token too many times. At the end an incorrect refresh token was store to local storage.

## Solution

Switching http provider to a default angular `HttpClient` allows us to use http interceptors and resolve token refreshing on their level. Cors will be solved on our backend.

By using multiple interceptors we can decouple logic for every backend. To be able to trigger a specific interceptor we use a http custom headers `x-api-resource`.

## Important

Please check if all http calls related to our backend are using correct interceptor.

## Test

- check entire authentication flow
- Wait min 30 seconds (temporary duration) if swap or reaching data from our server works
- Check if there is no logout after 30 seconds
- Check if swap works
- Check if user action in settings work (password change,...)
